### ISSUE TITLE
SyntaxEditor.SetAccessibility doesn't change the accessibility of inner getter/setter

### ISSUE THREAD

**Issue by mavasani** [dotnet/roslyn#2654]
```csharp
    public class ImplementsNestedGeneral
    {
        internal string Name
        {
            private get
            {
                return "name";
            }
        }
    }
```

Invoke `SyntaxEditor.SetAccessibility` on the property symbol for "Name" with below piece of fixer code to make an existing property effectively public:

```csharp
private async Task<Document> MakePublic(Document document, ISymbol symbolToChange, CancellationToken cancellationToken)
        {
            var editor = SymbolEditor.Create(document);

            await editor.EditAllDeclarationsAsync(symbolToChange, (docEditor, declaration) =>
            {
                docEditor.SetAccessibility(declaration, Accessibility.Public);
            }, cancellationToken);

            return editor.GetChangedDocuments().First();
        }
```

**Actual**: The returned declaration has the property's accessibility changed to `public`, but the getter method is still `private`, causing the property to still be semantically a private property.

```
    public class ImplementsNestedGeneral
    {
        public string Name
        {
            private get
            {
                return "name";
            }
        }
    }
```

Arguably, this is by design and the code fix author should deal explicitly with tweaking the property of the accessor. However, doing so would require the fixer to be split across languages, as there are different language specific restrictions on when a getter/setter can have explicit accessibility modifiers. The primary issues are:

1. In VB, getters for properties marked as `Readonly` cannot have accessibility modifiers, but otherwise they can.

2. There is no language independent way to find out if a Getter/Setter method has any accessibility modifiers or not. `SyntaxGenerator.GetAccessiblity` API helps determine the effective accessibility (http://index/Microsoft.CodeAnalysis.Workspaces/R/7b5a7f4f1e5e940a.html), but not the accessibility modifiers explicitly on the accessor.

In short, there is no language agnostic way of making an existing property declaration (with a more restrictive accessor accesibiltiy), semantically public by using the SyntaxGenerator APIs, without adding language specific components to the fixer.

**Comment by mavasani** [comment]
See skipped unit tests for FxCop rule CA1033 fixer added with #2590, which are affected by this bug. Skipped tests have been marked with this issue.

If this issue is resolved as by-design, I'll write language-specific fixers to address this scenario.

**Comment by Pilchie** [comment]
@mattwar what do you think about the `SyntaxGenerator` in this case?

**Comment by mattwar** [comment]
You can actually manipulate the accessors directly.   

```
var prop = ...;
var acc = gen.GetAccessor(prop, DeclarationKind.GetAccessor);
acc = gen.WithAccessibility(acc, Accessibility.Public)
prop = gen.WithAccessor(prop, acc)
```

I'm not sure what would be the right behavior for the SyntaxGenerator to do when WithAccessiblity is called on a property where the accessors have independent accessibilities declared. Likely, this is an advanced scenario.

**Comment by mavasani** [comment]
@mattwar Let me try this for my fixer/skipped tests. I am assuming this will handle cases where:

1. Accessibility modifiers are not valid on accessors (e.g. events).
2. Existing, more restrictive accessibility modifiers on accessors are removed.
3. No new modifiers are added if the original accessor had no explicit accessibility modifiers.

**Comment by mattwar** [comment]
@mavasani it will just do what you tell it to do.

**Comment by mattwar** [comment]
@mavasani  are you still looking for a change to this API?

**Comment by mavasani** [comment]
@mattwar I haven't had time to look at it yet. However, I think its safe to either resolve this as by-design for now OR move it to 1.1 milestone as a workaround exists (use language specific components in the fixer).
